pkg/trickster/prometheus/client.go

```diff
@@ -0,0 +1,55 @@
+package prometheus
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/Comcast/trickster/internal/cache/registration"
+	"github.com/Comcast/trickster/internal/config"
+	"github.com/Comcast/trickster/internal/proxy/origins/prometheus"
+)
+
+type processFunc func(*TricksterClient, http.ResponseWriter, *http.Request)
+
+type TricksterClient struct {
+	client       *prometheus.Client
+	apiPrefixMap map[string]processFunc
+}
+
+func NewTricksterClient(originConfig *config.OriginConfig) (*TricksterClient, error) {
+	c, err := registration.GetCache(originConfig.CacheName)
+	if err != nil {
+		return nil, err
+	}
+	t := &TricksterClient{
+		client: prometheus.NewClient(originConfig.Type, originConfig, c),
+		apiPrefixMap: map[string]processFunc{
+			originConfig.PathPrefix + originConfig.APIPath + "/query_range?": processQueryRange,
+			originConfig.PathPrefix + originConfig.APIPath + "/query?":       processQuery,
+			originConfig.PathPrefix + originConfig.APIPath + "/series?":      processSeries,
+		},
+	}
+	return t, nil
+}
+
+func (t *TricksterClient) Process(w http.ResponseWriter, r *http.Request) {
```

Comment:
Personally I'd prefer if this was more structured, the prometheus clients for example have Query/QueryRange etc. -- which is what clients using this as a library would expect.

Reply:
Agreed and implemented. That makes the code even cleaner.

Reply:
I've added another commit to this PR with 2 changes:
First, I implemented @jacksontj recommendations regarding explicit API calls.
Second, I attempted to make the library more independent by passing logger to the library as opposed to using the static logger.
The code is not completed and won't compile, but the overall idea is there.
I would like to get the feedback before proceeding with this rather a massive change.
Thanks!